dpnp/backend/kernels/dpnp_krnl_indexing.cpp

```diff
@@ -26,12 +26,46 @@
 #include <iostream>
 #include <list>
 #include <vector>
+#include <list>
 
 #include <dpnp_iface.hpp>
 #include "dpnp_fptr.hpp"
 #include "dpnpc_memory_adapter.hpp"
 #include "queue_sycl.hpp"
 
+template <typename _DataType1, typename _DataType2>
+class dpnp_choose_c_kernel;
+
+template <typename _DataType1, typename _DataType2>
+void dpnp_choose_c(void* result1, void* array1_in, std::vector<void*> choices1, size_t size)
```

Comment:
please do not use STL in the API. Change it as `void* choses, size_t choices_size`